mkosi/__init__.py

```diff
@@ -4126,28 +4127,27 @@ def secure_boot_sign(
         return
 
     with mount():
-        for path, _, filenames in os.walk(root / "efi"):
-            for i in filenames:
-                if not i.endswith(".efi") and not i.endswith(".EFI"):
-                    continue
+        for f in itertools.chain(efi_path.glob('**/*.efi'), efi_path.glob('**/*.EFI')):
```

Comment:
I'd prefer if we do not glob recursive everything here. I'd change this to just glob "*.efi" and call secure_boot_sign() multiple times if you need to sign in multiple directories